### ISSUE TITLE
Speedrun timer isn't displayed properly in Multiplayer

### ISSUE THREAD

**Issue by 0lhi** [Wohlstand/TheXTech#138]
**Describe the issue**

During the screensplit, the left half of the timer gets displayed twice. Only Player 1's inputs are shown, also twice.

**Recording**

![Scr_2021-06-10_22-12-05](https://user-images.githubusercontent.com/74732674/121593273-03785b80-ca3c-11eb-8026-0e48f0f5ffa9.gif)
![Scr_2021-06-10_22-45-43](https://user-images.githubusercontent.com/74732674/121594736-c0b78300-ca3d-11eb-9787-3c711c5d57bf.gif)

**Vanilla Recording**

Not that it matters, but it's the [same for SMBX2](https://user-images.githubusercontent.com/74732674/121594292-28210300-ca3d-11eb-9b01-7075b0c74a4a.gif).

**Additional context**

It would probably be best to make the Inputs red-transparent and green-transparent (maybe other colors if 4-Player ever becomes a thing), and have it be placed at the far-left/far-right of the splitscreen. If dedicated player-windows or two 800x600 screens, as described in #46, are implemented, it should always be on the left.

There also doesn't seem to much sense in displaying the life and coin icons twice? May also be something to take a look at, but not as critical as the other things.

**Comment by Wohlstand** [comment]
I really must render it after all viewports... Thanks for the catch! :fox_face: :+1:

**Comment by Wohlstand** [comment]
Also, if two-player mode, I probably should draw the second "controller" at the right side of the screen :eyes:

**Comment by Wohlstand** [comment]
I made the fix for this by moving the speed-run timer drawing into the front of both player viewports, so, it should be fine.

**Comment by 0lhi** [comment]
The Frame Counter is displayed on both Players still. Other than that, it all seems good.

The new Input Display is a very nice change 👍

Is it possible to enable it without the timer?

**Comment by Wohlstand** [comment]
> Is it possible to enable it without the timer?

Not yet, but a good idea, easy to implement :fox_face:

**Comment by Wohlstand** [comment]
At first, I'll make them be shown at v-screens to be seen at every player separately, that shouldn't confuse people who watching the video

**Comment by 0lhi** [comment]
It's not a big issue, but the new position of the time leans a bit heavy to the right. This is especially noticeable past the 1-hour-mark:
![Scr_2021-06-12_18-49-44](https://user-images.githubusercontent.com/74732674/121783478-1cede480-cbaf-11eb-8f7b-a4f6c70f8c25.gif)

Would it be possible to have the original one back? That way, the ratio of numbers is either 4-to-3 or 4-to-5, which is nicer to look at.

Alternatively, can it be made so that the position of the milliseconds is fixed, and the timer grows to the left? That would probably be even better, when it's not too complicated.

**Comment by 0lhi** [comment]
Timer position is improved. [[1]](https://user-images.githubusercontent.com/74732674/122902196-94113e80-d34e-11eb-9f52-9db523058a1a.png) [[2]](https://user-images.githubusercontent.com/74732674/122905477-8c9f6480-d351-11eb-9e86-82c1d9414550.png)


Colored controllers is a good idea! It needs some improvements:

![Scr_2021-06-22_11-41-05](https://user-images.githubusercontent.com/74732674/122902505-c9b62780-d34e-11eb-85ff-e662f5da38a6.png)

The position changes too fast. Player 1's controller is right, but his Inventory is still to the left. If the screen is unsplit, it should always stick to the left.

![Scr_2021-06-22_11-46-03](https://user-images.githubusercontent.com/74732674/122903195-755f7780-d34f-11eb-8740-91d405a23e0e.png)

The Color should be determined by Character and not Player. gameinfo.ini could be used to change it for custom packs. The free space in the middle can be used for a "P1", "P2" in the SMBX Font.

**Comment by Wohlstand** [comment]
What about the leave event, I changed it, so, please test it :fox_face: :+1: 
The colors of meta-controllers now depend on the playable character, however, I had to just mark the code as `todo` for the future, not hurry on that yet.

**Comment by 0lhi** [comment]
Much better! Mario's red could be a bit stronger, and Link's color should be the same as his Character block ([#c0a848](https://www.colorion.co/c0a848)).

**Comment by Wohlstand** [comment]
Done, if that fine now, I'll close this.

**Comment by 0lhi** [comment]
There's still the issue that the Controller positions change too fast.

![Scr_2021-06-23_14-45-11](https://user-images.githubusercontent.com/74732674/123098652-a7490a80-d431-11eb-83a6-74d541b1fd36.gif)

**Comment by Wohlstand** [comment]
I don't think this is an issue, or at least I made the solution is simple as things like transitions will require extra variables and counters, also to track their position state and make the change happening when one player will get farther than the other in specific pixels amount and don't switch them often in situations like you shown. That's the task for future work and improvements.

**Comment by 0lhi** [comment]
What do you mean by transitions?

I think it should be like this:

1. _If screen is split horizontally, place each player's contoller to their screen's bottom left._ ([Already the case](https://user-images.githubusercontent.com/74732674/123103360-35bf8b00-d436-11eb-9bbf-20c32542c41f.png).)

2. _If screen is split vertically, place left player's controller to the right, and right player's controller to the right._ (In practice the case.)

3. _If screen is unsplit, have P1's controller always on the left, P2's controller always on the right._

Would that really be hard to do?

**Comment by Wohlstand** [comment]
> If screen is unsplit, have P1's controller always on the left, P2's controller always on the right.

I did this initially, however, my idea was to make the controller of each player is displayed closer to the playable character itself, but yeh, I can comment out this logic and keep controllers being displayed at fixed positions.

**Comment by 0lhi** [comment]
Great! One more thing: Toad's Blue on the controller has to be brighter. Once that is done, this can be closed :)

**Comment by Wohlstand** [comment]
Can you specify the exact color level?
Right now it's next:
```cpp
r = 0.3f;
g = 0.3f;
b = 0.5f;
```

**Comment by 0lhi** [comment]
Not immediately. I can try around a bit and do the change myself later.

**Comment by Wohlstand** [comment]
these are OpenGL color levels between 0 and 1, simply multiply them with 255 to get the RGB value, and to get OpenGL color, divide the channel level with 255

**Comment by 0lhi** [comment]
```
r = 0.04f;
g = 0.43f;
b = 1.00f;
```

Leads to a good result ([#0b6dff](https://www.colorion.co/0b6dff), same as Toad's Switch Block):

![Scr_2021-06-23_16-32-00_(1)](https://user-images.githubusercontent.com/74732674/123115921-c56a3700-d440-11eb-9ea1-13d27ca5d01d.png)

**Comment by Wohlstand** [comment]
Done

**Comment by 0lhi** [comment]
Looks like all is done here 👍
